packages/shared/src/hooks/bookmark/useCreateBookmarkFolder.ts

```diff
@@ -0,0 +1,49 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { BookmarkFolder, createBookmarkFolder } from '../../graphql/bookmarks';
+import { useLogContext } from '../../contexts/LogContext';
+import { LogEvent } from '../../lib/log';
+import { useToastNotification } from '../useToastNotification';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+
+type CreateBookmarkFolderProps = Pick<BookmarkFolder, 'name' | 'icon'>;
+
+interface UseCreateBookmarkFolder {
+  isPending: boolean;
+  createFolder: (folder: CreateBookmarkFolderProps) => void;
+}
+
+export const useCreateBookmarkFolder = (): UseCreateBookmarkFolder => {
+  const { logEvent } = useLogContext();
+  const { displayToast } = useToastNotification();
+  const queryClient = useQueryClient();
+  const { mutateAsync, isPending } = useMutation({
+    mutationFn: createBookmarkFolder,
+  });
+
+  const createFolder = useCallback(
+    async (folder: CreateBookmarkFolderProps) => {
+      await mutateAsync(folder)
+        .then(async ({ id }) => {
+          logEvent({
+            event_name: LogEvent.CreateBookmarkFolder,
+            target_id: id,
+          });
+
+          const listQueryKey = generateQueryKey(RequestKey.BookmarkFolders);
+          await queryClient.invalidateQueries({ queryKey: listQueryKey });
```

Comment:
This more a few on optimistic updating and manipulating data.

Reply:
Do you think we should not fetch this anymore on creation?

Reply:
In general we prefer to do optimistic updates where possible.
In this case we know 100% what will happen right?
So we can take care of it and avoid having to re-fetch on invalidation.

Reply:
Updated ✔️